### PR TITLE
374 restyle bank payment section

### LINKED ISSUE #374: [FRONTEND] Restyle bank payment section
**Is your feature request related to a problem? Please describe.**

You will need to edit `client\src\components\composite\SignUpForm\Sections\PaymentSection.tsx`

Image

**BEFORE MERGING**

- [ ] Storybooks created where possible
- [ ] Tested with [nock](https://tanstack.com/query/v3/docs/framework/react/guides/testing) if data fetching
- [ ] Tests written for critical interactions
- [ ] PR Reviewed (For non-trivial changes)
- [ ] Changes tested after rebasing on master or merging in master (_hint_: `git fetch origin master:master`, then `git rebase master` or `git merge master`)
- [ ] All required PR checks passing

### PR DESCRIPTION
**Pull request by zlrkw11**
closes #374

### CODE REVIEW

**Comment by zlrkw11** [comment]
image
- default desktop view, i kept the part for 
image
(can remove if needed)

image
- mobile version

I assume title is waiting to be added when implemented

**Comment by zlrkw11** [comment]
I realized I forgot the copy button. Will implement it tommorow.

File: client/src/components/composite/SignUpForm/Sections/PaymentSection.tsx [hunk 1]
```diff
@@ -17,48 +17,52 @@ type PaymentSectionProps = { wantsBankTransfer: (newState: boolean) => void }
...
-        been confirmed by UASC.
-      </p>
-      <p>
-        You will still have access to your account, and are able to set your
-        login details as well as edit your personal information
-      </p>
+      <div className="flex gap-4">
+        <h4>1.</h4>
+        <div>
+          <h4 className="">Transfer payment amount to bank number:</h4>
```
Comment (bcho892) [inline client/src/components/composite/SignUpForm/Sections/PaymentSection.tsx:33]: We should not be hard coding these, use the queries from before

File: client/src/components/composite/SignUpForm/Sections/PaymentSection.tsx [hunk 2]
```diff
@@ -17,48 +17,52 @@ type PaymentSectionProps = { wantsBankTransfer: (newState: boolean) => void }
 const BankTransferSection = ({ wantsBankTransfer }: PaymentSectionProps) => {
   const navigate = useNavigate()
   const [{ currentUser }] = useAppData()
-  const { data: prices } = useMembershipPricesQuery()
-  const { data } = useBankPaymentDetailsQuery()
```
Comment (bcho892) [inline client/src/components/composite/SignUpForm/Sections/PaymentSection.tsx:21]: Why are these being deleted?

Comment (bcho892) [reply client/src/components/composite/SignUpForm/Sections/PaymentSection.tsx:21]: They are required to pay for membership

**Review by bcho892 — CHANGES_REQUESTED** [review]

File: client/src/components/composite/SignUpForm/Sections/PaymentSection.tsx [hunk 3]
```diff
@@ -17,48 +17,52 @@ type PaymentSectionProps = { wantsBankTransfer: (newState: boolean) => void }
...
-      <p>
-        Note you will not be able to make any bookings until your payment has
-        been confirmed by UASC.
-      </p>
-      <p>
-        You will still have access to your account, and are able to set your
-        login details as well as edit your personal information
-      </p>
+      <div className="flex gap-4">
+        <h4>1.</h4>
```
Comment (bcho892) [inline client/src/components/composite/SignUpForm/Sections/PaymentSection.tsx:31]: Should probably use an `ol` (ordered list) for this

**Comment by zlrkw11** [comment]
image
- copy button made
- ordered list used (spacing fixed inside of it using tailwind)
- added back the price props

**Comment by zlrkw11** [comment]
image
- mobile view

**Comment by zlrkw11** [comment]
notes: I wanna make mouse cursor to be default (not pointer) after the copy button is clicked.

File: client/src/components/composite/SignUpForm/Sections/PaymentSection.tsx [hunk 4]
```diff
@@ -11,54 +11,76 @@ import {
...
-   */
+  const { data } = useBankPaymentDetailsQuery()
   const requiredPrice = prices?.find(
     (price) => price.type === userMembershipDetails?.membershipType
   )
+  /**
+   * Use data fetched to find the correct price
+   */
+
+  const CopyButton = (text: any) => {
```
Comment (bcho892) [inline client/src/components/composite/SignUpForm/Sections/PaymentSection.tsx:31]: nit: gj on this but:
```suggestion
  const CopyButton = (text?: string) => {
```

File: client/src/components/composite/SignUpForm/Sections/PaymentSection.tsx [hunk 5]
```diff
@@ -11,54 +11,76 @@ import {
...
+    const handleOnclick = () => {
+      navigator.clipboard.writeText(text)
```
Comment (bcho892) [inline client/src/components/composite/SignUpForm/Sections/PaymentSection.tsx:33]: and
```suggestion
  try {
    await navigator.clipboard.writeText(text);
  } catch (error) {
    console.error(error.message);
}
```
its asynchronous

Comment (zlrkw11) [reply client/src/components/composite/SignUpForm/Sections/PaymentSection.tsx:33]: cheers bro

File: client/src/components/composite/SignUpForm/Sections/PaymentSection.tsx (hunk 4 above)
Comment (bcho892) [inline client/src/components/composite/SignUpForm/Sections/PaymentSection.tsx:31]:
```suggestion
  const CopyButton = ({text}:{text?:string}) => {
```

**Review by bcho892 — APPROVED** [review]
Good job on figuring out the copy!

**Review by bcho892 — CHANGES_REQUESTED** [review]
Screenshot_20240525-202127

Actually u may want to fix the order of this text, but otherwise looks good

**Comment by zlrkw11** [comment]
resolved

**Review by bcho892 — APPROVED** [review]

**Pull request merged**